yb-voyager/cmd/importData.go

```diff
@@ -798,6 +804,71 @@ func importData(importFileTasks []*ImportFileTask, errorPolicy importdata.ErrorP
 	fmt.Printf("\nImport data complete.\n")
 }
 
+// For a fresh start but non empty tables in tableList && OnPrimaryKeyConflict is set to IGNORE -> notify user
+func runPKConflictModeGuardrails(importFileTasks, pendingTasks, completedTasks []*ImportFileTask) error {
+	// in case of ERROR mode, no need to check for non-empty tables
+	// but for IGNORE or UPDATE(in future), we need to prompt user
+	if tconf.OnPrimaryKeyConflictAction == constants.PRIMARY_KEY_CONFLICT_ACTION_ERROR {
+		return nil
+	}
+
+	if !isTargetDBImporter(importerRole) {
+		return nil
+	}
+
+	if !isFreshStart(importFileTasks, pendingTasks, completedTasks) {
+		log.Infof("Not a fresh start, skipping primary key conflict mode check.")
+		return nil
+	}
+
+	pendingTablesList := importFileTasksToTableNameTuples(pendingTasks)
+	nonEmptyTables := tdb.GetNonEmptyTables(pendingTablesList)
+	if len(nonEmptyTables) == 0 {
+		return nil
+	}
+
+	nonPKTables := map[sqlname.NameTuple]bool{}
+	for _, table := range nonEmptyTables {
+		colList, err := tdb.GetPrimaryKeyColumns(table)
+		if err != nil {
+			return fmt.Errorf("failed to get primary key columns for table %s: %w", table.ForOutput(), err)
+		}
+		if len(colList) != 0 {
+			nonPKTables[table] = true
+		}
+	}
+
+	// all nonEmptyTables have no primary key columns
+	if len(nonPKTables) == len(nonEmptyTables) {
+		log.Infof("All tables in %v have no primary key columns, skipping primary key conflict mode check.", sqlname.NameTupleListToStrings(nonEmptyTables))
+		return nil
+	}
+
+	pkTables := lo.Filter(nonEmptyTables, func(table sqlname.NameTuple, _ int) bool {
+		return !nonPKTables[table]
+	})
```

Comment:
I was slightly confused by this block. 
How about we: 
- get all the nonEmptyTablesHavingPK. 
- if nonEmptyTablesHavingPK == 0, return (same as checking if all are non-pk)
- for all the nonEmptyTablesHavingPK, prompt.

We can avoid getting nonPk tables first and then flipping it to pkTables